File: app/src/main/java/com/engineeringforyou/basesite/presentation/searchsite/views/SearchSiteView.java
package com.engineeringforyou.basesite.presentation.searchsite.views;

import android.database.Cursor;
import android.support.annotation.StringRes;

public interface SearchSiteView {
    void setOperator(int operatorIndex);

    void showError(@StringRes int textRes);

    void showResult(@StringRes int textRes);

    void showProgress();

    void hideProgress();

    void toSiteInfo(Cursor cursor);

    void toSiteChoice(Cursor cursor, int count);
}

File: app/src/main/java/com/engineeringforyou/basesite/presentation/searchsite/views/SearchSiteView.kt
package com.engineeringforyou.basesite.presentation.searchsite.views

import android.support.annotation.StringRes
import com.engineeringforyou.basesite.models.Site

interface SearchSiteView {

    fun setOperator(operatorIndex: Int)

    fun showError(@StringRes textRes: Int)

    fun hideError()

    fun showProgress()

    fun hideProgress()

    fun toSiteInfo(site: Site)

    fun toSiteChoice(list: List<Site>)

    fun openMap()

}